e2e-tests/helpers/boundary-cli/connect.js

```diff
@@ -11,47 +11,115 @@ import { exec } from 'node:child_process';
  * @param {string} targetId ID of the target to be connected to
  * @param {string} sshUser User to be used for the ssh connection
  * @param {string} sshKeyPath Path to the ssh key to be used for the ssh connection
+ * @param {boolean} ignoreErrors (Optional) boolean to ignore errors
```

Comment:
What error do you get from canceling a session? Is it from the act of killing the process?

Reply:
During testing, most of the time I would see something like this
```
Error: Command failed: boundary connect ssh -target-id=tssh_CkgJefNJqW -- -T -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes
Connection to 127.0.0.1 closed by remote host.

    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1105:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:519:28)
    at Pipe.<anonymous> (node:net:338:12)
    at Pipe.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 255,
  killed: false,
  signal: null,
  cmd: 'boundary connect ssh -target-id=tssh_CkgJefNJqW -- -T -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes'
```
where `killed: false`.

Reply:
Ah i forget that you're also doing the ssh connection in the same command so canceling it also kills the connection.

I wonder if it's worth just checking if the error message has the `Connection to 127.0.0.1 closed by remote host` as that's one you can just ignore it seems like? If there's other scenarios where you want to ignore the errors though I think it's also fine to keep it as it is

Reply:
I was a little nervous doing that as I think you can also get that error message if something wasn't set up correctly (like incorrect credentials).

Reply:
You wouldn't get a permissions denied error instead? I assumed if it didn't connect properly you'd get a real error since it wouldn't have connected yet to be able to get the `closed by remote host`

Reply:
That's still a different error from `closed by remote host` isn't it? But either way, I think it's fine to just be explicit about when we expect it to "fail" here by using that flag

Reply:
Ah yup. Good call. Reading hard.

But yea, agreed, I think we can keep as-is for now.